tephra-hbase-compat-2.0/src/main/java/org/apache/tephra/hbase/TransactionAwareHTable.java

```diff
@@ -0,0 +1,729 @@
+                        NavigableMap<byte[], byte[]> familyColumns = result.getFamilyMap(family);
+                        for (Map.Entry<byte[], byte[]> column : familyColumns.entrySet()) {
+                            txDelete.addColumns(family, column.getKey(), transactionTimestamp);
+                            addToChangeSet(deleteRow, family, column.getKey());
+                        }
+                    }
+                } else {
+                    for (Cell value : entries) {
+                        txDelete.addColumn(CellUtil.cloneFamily(value),
+                            CellUtil.cloneQualifier(value), transactionTimestamp);
+                        addToChangeSet(deleteRow, CellUtil.cloneFamily(value),
+                            CellUtil.cloneQualifier(value));
+                    }
+                }
+            }
+        }
+        for (Map.Entry<String, byte[]> entry : delete.getAttributesMap().entrySet()) {
+            txDelete.setAttribute(entry.getKey(), entry.getValue());
+        }
+        txDelete.setDurability(delete.getDurability());
+        addToOperation(txDelete, tx);
+        return txDelete;
+    }
+
+    private List<? extends Row> transactionalizeActions(List<? extends Row> actions)
+            throws IOException {
+        List<Row> transactionalizedActions = new ArrayList<>(actions.size());
+        for (Row action : actions) {
+            if (action instanceof Get) {
+                transactionalizedActions.add(transactionalizeAction((Get) action));
+            } else if (action instanceof Put) {
+                transactionalizedActions.add(transactionalizeAction((Put) action));
+            } else if (action instanceof Delete) {
+                transactionalizedActions.add(transactionalizeAction((Delete) action));
+            } else {
+                transactionalizedActions.add(action);
+            }
+        }
+        return transactionalizedActions;
+    }
+
+    public void addToOperation(OperationWithAttributes op, Transaction tx) throws IOException {
+        op.setAttribute(TxConstants.TX_OPERATION_ATTRIBUTE_KEY, txCodec.encode(tx));
+    }
+
+    protected void makeRollbackOperation(Delete delete) {
+        delete.setAttribute(TxConstants.TX_ROLLBACK_ATTRIBUTE_KEY, new byte[0]);
+    }
+
+    @Override
+    public TableDescriptor getDescriptor() throws IOException {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public boolean[] exists(List<Get> gets) throws IOException {
+        if (tx == null) {
+            throw new IOException("Transaction not started");
+        }
+        List<Get> transactionalizedGets = new ArrayList<>(gets.size());
+        for (Get get : gets) {
+            transactionalizedGets.add(transactionalizeAction(get));
+        }
+        return hTable.exists(transactionalizedGets);
+    }
+
+    @Override
+    public boolean checkAndPut(byte[] row, byte[] family, byte[] qualifier, CompareOperator op,
```

Comment:
Would be good to move the `checkAnd...` methods here alongside the `checkAnd...` earlier in the file.